main/midnight/src/ui/uibookmenu.cpp

```diff
@@ -138,9 +138,13 @@ void uibookmenu::Show()
 
 void uibookmenu::Close()
 {
-    //
+    // get parent so we can send events to 
+    // it once we do remove child
+    uipanel* menuParent = parent;
+
+    // remove child, this sets parent to null
     parent->removeChild(this);
     
     // give the parent events back
-    parent->resumeEvents();
+    menuParent->resumeEvents();
```

Comment:
Was getting null pointer exception error here. line 146 ends up setting parent to null when removing child.

Reply:
Following calls, this ends up calling:
CCNode.cpp::removeChild

that one in turn calls:
CCNode.cpp::detachChild

And that one does:
   // set parent nil at the end
    child->setParent(nullptr);